### ISSUE TITLE
Offset unit autoconversion induces incorrect "degC days"

### ISSUE THREAD

**Issue by aulemahal** [Ouranosinc/xclim#607]
* xclim version: since 0.14
* Python version: all
* Operating System: all

### Description

When parsing a string expression containing the "degC days" unit, our current setup of pint in xclim converts `degC` to `273.15 K` automatically, which results in completely wrong quantities.

### What I Did

```python3
import xclim as xc
xc.units.convert_units_to('10 degC days', 'K days') -> 283.15
```
This is problematic for #606, which uses a deg-day threshold.

This error comes from:
 https://github.com/Ouranosinc/xclim/blob/ff021efe643817a038da3464bc92288f71f0dd59/xclim/core/units.py#L34

Is this line really needed? I'm not sure of the utility. I'll test removing it.

If we set that to `False`, the line above doesn't return `10` as we could expect, but rather raises:
 ```
OffsetUnitCalculusError: Ambiguous operation with offset unit (degree_Celsius, ).
```
Which I prefer, at least it doesn't make wrong results silently! This would force users to provide the values in `K days` instead, which is better defined.

What do you think?

**Comment by tlogan2000** [comment]
@aulemahal similar but completely unrelated : 

I recently had some strange results with a dataset I created with both absolute future horizon means but also relative deltas.  Original data was in kelvin so delta units remained in the attrs (e.g. +5 degK) ...  I blindly applied a conversion to degC on all data_vars in the dataset and ended with degC deltas of ~-265 to -270 etc.   Basically changing units on a relative temperature change is a not straight forward or even possible ...  Not sure we can handle all of these type of cases but there is a gotcha there

**Comment by aulemahal** [comment]
Pint has a `delta_degree_Celsius` unit, but it's quite long to type...

**Comment by huard** [comment]
I believe delta_degree_Celsius is the correct way to express degree days. Agree it's long, but we could create a shorter alias for it.
Untested:
```python
units.define("degdayC = delta_degree_Celsius * day")
```

**Comment by aulemahal** [comment]
Ok, so a lot of tests  will fail with this option turned off. This is because we use `units.parse_expression` to transform strings into pint `Quantity`. That function parses each element has a _multiplication_. Meaning `12 degC` is understood as `12 * 1 degC`. which is illdefined.

## Proposition
Change our string parsing system:
```python3
def parse_string(s):
    mag, uni = s.split(' ', max_split=1)
    mag = float(mag)
   return units.Quantity(mag, units=units.parse_units(uni))
```
Where `units` is the  `UnitsRegistry` instance (`xclim.core.units.units`).

This understands `degC days` as `delta_degree_Celsius * day`. I feel it is closer to what we expect when writing quantities in strings.

**Comment by tlogan2000** [comment]
> I believe delta_degree_Celsius is the correct way to express degree days. Agree it's long, but we could create a shorter alias for it.
> Untested:
> 
> ```python
> units.define("ddC = degDC = degdayC = delta_degree_Celsius * day")
> ```

+1 for this idea ...  but why  `* day`?    isn't a degree day literally either  `degC - degC` or `degK - degK` for each day  in each case delta_degree_Celsius is an appropriate unit

**Comment by huard** [comment]
Euh... maybe I'm wrong., but I think we need to describe units in such a way that we can for example compute delta_degree_Celsius at hourly time scales, then convert to daily. 
```delta_degree_Celsius hours -> delta_degree_Celsius days```. 

`24 delta_degree_Celsius hours == 1 delta_degree_Celsius days`

**Comment by tlogan2000** [comment]
> Euh... maybe I'm wrong., but I think we need to describe units in such a way that we can for example compute delta_degree_Celsius at hourly time scales, then convert to daily.
> `delta_degree_Celsius hours -> delta_degree_Celsius days`.
> 
> `24 delta_degree_Celsius hours == 1 delta_degree_Celsius days`

of course... makes sense

**Comment by aulemahal** [comment]
Closed by #608, I mis-referred the issue.
